### PR TITLE
home loaders

### PR DESCRIPTION
**Pull request by ice-orion**
Adds skeleton loaders to the Home page

### CODE REVIEW

File: src/store/modules/Achievements/reducer/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {AchievementsActions} from '@store/modules/Achievements/actions';
 import produce from 'immer';
 
 interface State {
-  tasks: Task[];
+  tasks: Task[] | null;
```
Comment (ice-atlas) [inline src/store/modules/Achievements/reducer/index.ts:10]: Why we need `null` value? Is it not enough empty array as it was before?

File: src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts [hunk 2]
```diff
@@ -10,16 +10,16 @@ import {useDispatch, useSelector} from 'react-redux';
 export function useTasks() {
   const dispatch = useDispatch();
 
-  const [currentTasks, setCurrentTasks] = useState<Task[]>([]);
+  const [currentTasks, setCurrentTasks] = useState<Task[] | null>([]);
 
   const tasks: ReturnType<typeof AchievementsSelectors.getTasks> = useSelector(
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts]: Remove pls `ReturnType<typeof AchievementsSelectors.getTasks>`

File: src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts [hunk 3]
```diff
@@ -10,16 +10,16 @@ import {useDispatch, useSelector} from 'react-redux';
 export function useTasks() {
   const dispatch = useDispatch();
 
-  const [currentTasks, setCurrentTasks] = useState<Task[]>([]);
+  const [currentTasks, setCurrentTasks] = useState<Task[] | null>([]);
```
Comment (ice-atlas) [inline src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts]: We do not need `<Task[] | null>` here if we place it after `useSelector` 
AND use as
`const [currentTasks, setCurrentTasks] = useState(tasks);`
(easier to maintain types)

File: src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts (hunk 2 above)
Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts]: done

File: src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts (hunk 3 above)
Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Tasks/hooks/useTasks.ts]: agree, done

File: src/store/modules/Achievements/reducer/index.ts (hunk 1 above)
Comment (ice-orion) [reply src/store/modules/Achievements/reducer/index.ts:10]: To distinguish "not loaded" state from the state when the data is returned but the array is empty.
In general, it looks better to set `null` as initial state - this way we won't mislead users with incorrect values (e.g. when we're setting 0 as default value for referrals number before we even loaded data).

Comment (ice-atlas) [reply src/store/modules/Achievements/reducer/index.ts:10]: In terms of data it is OK to have 0 or whatever while data loading, it is more about UI state (activityIndicator or smth).
I agree here with you, it is impossible to distinguish difference between default empty array when data not loaded yet, and loaded but just empty array (hasMore: false). But is it possible to have empty tasks array if it loaded?

Comment (ice-orion) [reply src/store/modules/Achievements/reducer/index.ts:10]: Discussed, decided to leave it as is for now

**Review by ice-atlas — APPROVED** [review]

File: src/screens/HomeFlow/Home/components/Overview/components/AdoptionCard/index.tsx [hunk 4]
```diff
@@ -171,58 +172,60 @@ export const AdoptionCard = ({isCollapsed, onPress}: AdoptionCardProps) => {
     [activeIndex, sharedItems, adoption?.milestones, isCollapsed, onPress],
   );
 
+  if (!isSplashHidden) {
```
Comment (ice-hector) [inline src/screens/HomeFlow/Home/components/Overview/components/AdoptionCard/index.tsx:175]: Looks like a good idea, does it optimize splash animation performance on launch?

Comment (ice-orion) [reply src/screens/HomeFlow/Home/components/Overview/components/AdoptionCard/index.tsx:175]: This logic was here before, but only a part of the card was not rendered if splash is visible. Now the whole card is not rendered until the splash is hidden.
And yea, the reason for this was that on android splash animation was played with hanging

**Review by ice-aiolos — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Pull request merged**